docs/reference/modules/node.asciidoc

```diff
@@ -1,32 +1,223 @@
+
+To create a standalone data node:
+
+```yaml
+node.master: false
+node.data: true
+```
+
+NOTE: It is not enough to just set `node.data: true`. You must
+disable the master aspect, which is enabled by default.
+
+[float]
+[[client]]
+===== Client Node
+
+If you take away the ability to be able to handle master
+duties and take away the ability to hold data, then you are left
+with a node that can only route requests.
+
+Standalone client nodes provide large scale clusters a few
+benefits, by offloading some of the overhead associated with
+federated search and even bulk indexing. Client nodes join the
+cluster and receive the full cluster state, like every other
+node, and they use the cluster state to route requests to the
+appropriate place(s). In terms of searching and indexing, this
+is known as federated search and federated indexing.
+
+Unlike master nodes and data nodes, the entire purpose of
+client nodes is to handle and route requests. This means that it
+is good practice to send them any type of request, although they
+are generally expected to be used for data-related requests.
+
+The benefit of the client nodes should not be overstated though.
+Client nodes benefit other nodes by offloading the federated
+aspect of the request (e.g., the memory burden associated with
+federating a large aggregation). The routing aspect is a
+convenience, but the focus on any node should be on the burden
+that they offload from other nodes.
+
+To create a standalone data node:
+
+```yaml
+node.client: true
+```
+
+NOTE: It _is_ enough to just set `node.client: true`. This will
+overrule `node.master` and `node.data` settings.
+
+[float]
+[[node-configuration]]
+===== Node Configuration
+
+The above descriptions noted how to setup nodes to be master,
+data, or client nodes, but this section discusses some
+universal node settings that should be considered with any
+deployment.
+
+[float]
+[[node-configuration-disk]]
+====== Disk Configuration
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+|Setting |Default |Description
+|`path.data` |`${ES_HOME}/data` |The place to store data (and metadata) for the
+node. This should be set uniquely per node.
+|`node.max_local_storage_nodes` |`50` |The number of nodes that can
+share the same `path.data` directory.
+|=======================================================================
+
+Every node should set its `path.data`, which, in the case of
+non-data nodes, allows the storage of metadata associated with
+the cluster. For data nodes, this is naturally where the data
+is stored.
+
+Whether you set the `path.data` or not, nodes can share
+the same data directory without interferring with other
+nodes, as long as you _always start them in the exact same
+order every time_. This can be a problem if you start
+multiple nodes, of varying types, on the same machine
+using an arbitrary order.
+
+To prevent `path.data` from being unexpectedly shared by
```

Comment:
The more important reason for max_local_storage_nodes is that you don't want to find you have two nodes running instead of just one.  Shards would be divided between `nodes/0` and `nodes/1`, then when you restart the box with just one node, you'd be missing shards.